// @webstudio-is/sdk-components-react/lib/time.js
import { jsx as S } from "react/jsx-runtime";
import { forwardRef as A } from "react";
const N = [
  "af",
  "am",
  "ar",
  "az",
  "be",
  "bg",
  "bn",
  "bs",
  "ca",
  "cs",
  "cy",
  "da",
  "de",
  "el",
  "en",
  "es",
  "et",
  "eu",
  "fa",
  "fi",
  "fr",
  "ga",
  "gl",
  "gu",
  "he",
  "hi",
  "hr",
  "hu",
  "hy",
  "id",
  "is",
  "it",
  "ja",
  "ka",
  "kk",
  "km",
  "kn",
  "ko",
  "ky",
  "lb",
  "lt",
  "lv",
  "mk",
  "ml",
  "mn",
  "mr",
  "ms",
  "mt",
  "nb",
  "nl",
  "nn",
  "pl",
  "pt",
  "ro",
  "ru",
  "si",
  "sk",
  "sl",
  "sq",
  "sr",
  "sv",
  "sw",
  "ta",
  "te",
  "th",
  "tr",
  "uk",
  "ur",
  "uz",
  "vi",
  "zh"
], g = [
  "AF",
  "AL",
  "DZ",
  "AS",
  "AD",
  "AO",
  "AI",
  "AQ",
  "AG",
  "AR",
  "AM",
  "AW",
  "AU",
  "AT",
  "AZ",
  "BS",
  "BH",
  "BD",
  "BB",
  "BY",
  "BE",
  "BZ",
  "BJ",
  "BM",
  "BT",
  "BO",
  "BA",
  "BW",
  "BR",
  "BN",
  "BG",
  "BF",
  "BI",
  "CV",
  "KH",
  "CM",
  "CA",
  "KY",
  "CF",
  "TD",
  "CL",
  "CN",
  "CO",
  "KM",
  "CG",
  "CD",
  "CR",
  "HR",
  "CU",
  "CY",
  "CZ",
  "DK",
  "DJ",
  "DM",
  "DO",
  "EC",
  "EG",
  "SV",
  "GQ",
  "ER",
  "EE",
  "SZ",
  "ET",
  "FJ",
  "FI",
  "FR",
  "GA",
  "GM",
  "GE",
  "DE",
  "GH",
  "GR",
  "GD",
  "GT",
  "GN",
  "GW",
  "GY",
  "HT",
  "HN",
  "HU",
  "IS",
  "IN",
  "ID",
  "IR",
  "IQ",
  "IE",
  "IL",
  "IT",
  "JM",
  "JP",
  "JO",
  "KZ",
  "KE",
  "KI",
  "KP",
  "KR",
  "KW",
  "KG",
  "LA",
  "LV",
  "LB",
  "LS",
  "LR",
  "LY",
  "LI",
  "LT",
  "LU",
  "MG",
  "MW",
  "MY",
  "MV",
  "ML",
  "MT",
  "MH",
  "MR",
  "MU",
  "MX",
  "FM",
  "MD",
  "MC",
  "MN",
  "ME",
  "MA",
  "MZ",
  "MM",
  "NA",
  "NR",
  "NP",
  "NL",
  "NZ",
  "NI",
  "NE",
  "NG",
  "NO",
  "OM",
  "PK",
  "PW",
  "PA",
  "PG",
  "PY",
  "PE",
  "PH",
  "PL",
  "PT",
  "QA",
  "RO",
  "RU",
  "RW",
  "KN",
  "LC",
  "VC",
  "WS",
  "SM",
  "ST",
  "SA",
  "SN",
  "RS",
  "SC",
  "SL",
  "SG",
  "SK",
  "SI",
  "SB",
  "SO",
  "ZA",
  "SS",
  "ES",
  "LK",
  "SD",
  "SR",
  "SE",
  "CH",
  "SY",
  "TW",
  "TJ",
  "TZ",
  "TH",
  "TL",
  "TG",
  "TO",
  "TT",
  "TN",
  "TR",
  "TM",
  "TV",
  "UG",
  "UA",
  "AE",
  "GB",
  "US",
  "UY",
  "UZ",
  "VU",
  "VA",
  "VE",
  "VN",
  "YE",
  "ZM",
  "ZW"
], f = "dateTime attribute is not set", E = "", c = "en", u = "GB", G = "medium", I = "none", L = (t) => N.includes(t) ? t : c, h = (t) => g.includes(t) ? t : u, B = (t) => {
  if (["full", "long", "medium", "short"].includes(t))
    return t;
}, R = (t) => {
  if (["full", "long", "medium", "short"].includes(t))
    return t;
}, U = (t) => {
  if (t === "")
    return;
  let e = new Date(t);
  if (Number.isNaN(e.getTime()) === !1)
    return e;
  if (/^\d+$/.test(t)) {
    let n = Number(t);
    t.length === 10 && (n *= 1e3), e = new Date(n);
  }
  if (Number.isNaN(e.getTime()) === !1)
    return e;
}, Y = (t, e, n = "en-US") => {
  const s = (r, o = 2) => String(r).padStart(o, "0"), D = new Intl.DateTimeFormat(n, {
    weekday: "long"
  }).format(t), T = new Intl.DateTimeFormat(n, {
    weekday: "short"
  }).format(t), l = new Intl.DateTimeFormat(n, {
    month: "long"
  }).format(t), M = new Intl.DateTimeFormat(n, {
    month: "short"
  }).format(t), m = {
    YYYY: t.getFullYear(),
    YY: String(t.getFullYear()).slice(-2),
    MMMM: l,
    MMM: M,
    MM: s(t.getMonth() + 1),
    M: t.getMonth() + 1,
    DDDD: D,
    DDD: T,
    DD: s(t.getDate()),
    D: t.getDate(),
    HH: s(t.getHours()),
    H: t.getHours(),
    mm: s(t.getMinutes()),
    m: t.getMinutes(),
    ss: s(t.getSeconds()),
    s: t.getSeconds()
  }, a = Object.keys(m).sort((r, o) => o.length - r.length), i = new RegExp(`\\b(${a.join("|")})\\b`, "g");
  return e.replace(i, (r) => String(m[r]));
}, H = A(
  ({
    language: t = c,
    country: e = u,
    dateStyle: n = G,
    timeStyle: s = I,
    format: D,
    // native html attribute in react style
    dateTime: T = f,
    ...l
  }, M) => {
    const m = `${L(t)}-${h(
      e
    )}`, a = {
      dateStyle: B(n),
      timeStyle: R(s)
    }, i = T === null ? E : T.toString(), r = U(i);
    let o = i;
    if (r)
      if (D)
        try {
          o = Y(r, D, m);
        } catch {
        }
      else
        try {
          o = new Intl.DateTimeFormat(m, a).format(r);
        } catch {
        }
    return /* @__PURE__ */ S("time", { ref: M, dateTime: i, ...l, children: o });
  }
);
export {
  H as Time
};
